xarray/tests/test_groupby.py

```diff
@@ -2551,54 +2551,93 @@ def func(arg1, arg2, arg3=0.0):
         assert_identical(expected, actual)
 
 
-def test_groupby_cumsum() -> None:
-    ds = xr.Dataset(
-        {"foo": (("x",), [7, 3, 1, 1, 1, 1, 1])},
-        coords={"x": [0, 1, 2, 3, 4, 5, 6], "group_id": ("x", [0, 0, 1, 1, 2, 2, 2])},
-    )
-    actual = ds.groupby("group_id").cumsum(dim="x")
-    expected = xr.Dataset(
-        {
-            "foo": (("x",), [7, 10, 1, 2, 1, 2, 3]),
-        },
-        coords={
-            "x": [0, 1, 2, 3, 4, 5, 6],
-            "group_id": ds.group_id,
-        },
-    )
-    # TODO: Remove drop_vars when GH6528 is fixed
-    # when Dataset.cumsum propagates indexes, and the group variable?
-    assert_identical(expected.drop_vars(["x", "group_id"]), actual)
-
-    actual = ds.foo.groupby("group_id").cumsum(dim="x")
-    expected.coords["group_id"] = ds.group_id
-    expected.coords["x"] = np.arange(7)
-    assert_identical(expected.foo, actual)
-
+@pytest.mark.parametrize(
+    "method, expected_array, use_flox, use_dask",
```

Comment:
The `use_dask` here should mean grouping a dask array by a numpy array. That will work always.